it/pages/VatDetailsPageSpec.scala

```diff
@@ -41,15 +45,56 @@ class VatDetailsPageSpec extends IntegrationBaseSpec {
 
     "the user is authenticated" should {
 
-      "return 200" in new Test {
+      "return 200 and 'View returns deadline'" in new Test {
```

Comment:
Can you state 'and submit return feature switch is on' or something similar for these tests, and add (just one) scenario for when switch is off to prove the existing logic still stands. I know you tested that in the controller spec but if you're doing these scenarios at integration level you may as well add one with the switch off too